### PR TITLE
Fix already borrowed error in webdriver

### LINKED ISSUE #23535: DomRefCell<T> already borrowed: BorrowMutError when handling execution of a script in WebDriver
When processing an execute script command with `window.open()` as the target script (`curl -v -X POST -d '{"script": "window.open()"}' "http://localhost:7002/session/$SESSIONID/execute/sync"`), the WebDriver server panics with the following error:

`DomRefCell<T> already borrowed: BorrowMutError (thread ScriptThread PipelineId { namespace_id: PipelineNamespaceId(1), index: PipelineIndex(1) }, at src/libcore/result.rs:999)`

This is where the script execution is handled: https://github.com/servo/servo/blob/a634f0502417421c78f3d05e493a12f0b19d83a0/components/script/webdriver_handlers.rs#L89

### PR DESCRIPTION
**Pull request by georgeroman**
---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #23535

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/23544)

### CODE REVIEW

File: components/script/script_thread.rs [hunk 1]
```diff
@@ -1833,6 +1833,18 @@ impl ScriptThread {
...
+            WebDriverScriptCommand::ExecuteScript(script, reply) => {
+                let window = { self.documents.borrow().find_window(pipeline_id) };
+                return webdriver_handlers::handle_execute_script(window, script, reply);
+            },
+            WebDriverScriptCommand::ExecuteAsyncScript(script, reply) => {
+                let window = { self.documents.borrow().find_window(pipeline_id) };
+                return webdriver_handlers::handle_execute_async_script(window, script, reply);
+            },
+            _ => (),
+        }
```
Comment (asajeffrey) [inline components/script/script_thread.rs]: A bit odd that we're handling these messages differently than the others? Is it worth making a similar change to the other message handles? If not, we should at least comment why these messages need handled differently than the others.

Comment (georgeroman) [reply components/script/script_thread.rs]: Yeah, that's true, I added a comment explaining the difference.

**Comment by jdm** [comment]
@bors-servo retry
* #23290

**Pull request merged**